### PR TITLE
Allow receiving an external tree, drop python 3.7

### PR DESCRIPTION
**Pull request by croqaz**
A small performance improvement: allow receiving an external tree into extruct.
Also bump the default Python version to 3.9, hope that's OK.

### CODE REVIEW

**Comment by lopuhin** [comment]
@croqaz thanks for the PR! Do you mind updating supported versions in https://github.com/scrapinghub/extruct/blob/9453e4348e06509e7716ada3de585cddfe2ad0d4/setup.py#L67 -- and also having a look at the CI, it seems we also run 3.7 and 3.11 on this branch (and then we don't need to drop 3.7 I guess, although I would not mind at all).

**Comment by croqaz** [comment]
According to https://devguide.python.org/versions/ at this point in time, 3.7 is dead, and 3.8 has a bit over a year. Looks safe to remove 3.7.

**Comment by lopuhin** [comment]
Sounds great! Do you mind also mentioning this feature in the README? We don't build docs here so it would be useful to mention it there.

File: tests/test_extruct_uniform.py [hunk 1]
```diff
@@ -13,20 +14,24 @@ class TestFlatten(unittest.TestCase):
 
     def test_microdata(self):
         body = get_testdata("schema.org", "CreativeWork.001.html")
+        tree = parse_html(body, encoding="UTF-8")
         expected = json.loads(
             get_testdata("schema.org", "CreativeWork_flat.001.json").decode("UTF-8")
         )
-        data = extruct.extract(body, uniform=True)
+        data = extruct.extract(tree, uniform=True, syntaxes=["microdata"])
```
[the remark this answers is not in the document]
Comment (croqaz) [reply tests/test_extruct_uniform.py]: I tried... but here we are using more of a unittest method, even if we seem to use pytest here in there.
So I would like to keep all the tests as they are, not rewrite everything and I made a common function to load the HTML and tree and use both for these tests.

[the remark this answers is not in the document]
Comment (croqaz) [reply tests/test_extruct_uniform.py]: What I mean is that I haven't found a good way to create Pytest-like fixtures in unittest, so I extracted as much common code as possible to be able to test both HTML and parsed tree in here.

**Review by lopuhin — APPROVED** [review]
Looks great, thanks @croqaz 👍

**Pull request merged**